openstack_controller/datadog_checks/openstack_controller/api/api_rest.py

```diff
@@ -281,11 +281,13 @@ def get_network_agents(self):
 
     def get_network_networks(self, project_id):
         params = {'project_id': project_id}
-        response = self.http.get(
-            '{}/v2.0/networks'.format(self._catalog.get_endpoint_by_type(Component.Types.NETWORK.value)), params=params
+        return self.make_paginated_request(
```

Comment:
nit comment, definitely not related to this particular PR so feel free to ignore: I think it would be better to `yield` the results from `make_paginated_request` without accumulating all the results in a single list. This would be more efficient memory wise in case you have a lot of pages because you would not need to store them all at once in a giant list, however I don't know if can happen a lot in your cases

Reply:
I agree with what @FlorentClarret proposes and I think that in some use cases it can be an improvement. However, I would do it in a future PR different from this one, because I think we can apply it in the same way in some other parts of our code. What do you think?

Reply:
Yes totally makes sense! I did not mean to introduce this in this PR, it was just a thought 🙂

Reply:
Agreed.